experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -463,3 +500,130 @@ function Checkpointer.get_model_prog_state(sim::ClimaLandSimulation)
+    z_int = Interfacer.get_field(atmos_sim, Val(:height_int))
+    uₕ_int = Interfacer.get_field(atmos_sim, Val(:uv_int))
+    thermo_state_int = Interfacer.get_field(atmos_sim, Val(:thermo_state_int))
+
+    # We use parent here because of the mismatch between the lowest atmos level
+    #  and the boundary space, even though the horizontal grids are the same.
+    @assert axes(coupled_atmos.u).grid == axes(uₕ_int).grid.full_grid.horizontal_grid
+    @assert axes(coupled_atmos.thermal_state).grid == axes(thermo_state_int).grid.full_grid.horizontal_grid
+    @assert axes(coupled_atmos.h).grid == axes(z_int).grid.full_grid.horizontal_grid
+    parent(coupled_atmos.u) .= parent(uₕ_int)
+    parent(coupled_atmos.thermal_state) .= parent(thermo_state_int)
+    parent(coupled_atmos.h) .= parent(z_int)
+
+    # set the same atmosphere state for all sub-components
+    @assert sim.model.soil.boundary_conditions.top.atmos ===
+            sim.model.canopy.boundary_conditions.atmos ===
+            sim.model.snow.boundary_conditions.atmos ===
+            coupled_atmos
+
+    # get area fraction (min = 0, max = 1)
+    area_fraction = Interfacer.get_field(sim, Val(:area_fraction))
+    # get area mask [0, 1], where area_mask = 1 if area_fraction > 0
+    area_mask = Utilities.binary_mask.(area_fraction)
+
+    Y, p, t = sim.integrator.u, sim.integrator.p, sim.integrator.t
+
+    # update `csf` in-place using this model's flux calculation
+    fluxes = coupler_land_turbulent_fluxes!(p, sim.model, Y, t, coupled_atmos)
+    (; F_turb_ρτxz, F_turb_ρτyz, F_shf, F_lhf, F_turb_moisture) = fluxes
+
+    # add the flux contributing from this surface to the coupler field
+    # note that the fluxes are area-weighted, so if a surface model is
+    #  not present at this point, the fluxes are zero
+    @. csf.F_turb_ρτxz += F_turb_ρτxz * area_fraction * area_mask
+    @. csf.F_turb_ρτyz += F_turb_ρτyz * area_fraction * area_mask
+    @. csf.F_turb_energy += (F_shf .+ F_lhf) * area_fraction * area_mask
+    @. csf.F_turb_moisture += F_turb_moisture * area_fraction * area_mask
+    return nothing
+end
+
+"""
+    coupler_land_turbulent_fluxes(model, Y, p, t, coupled_atmos)
+
+Compute the turbulent fluxes for the integrated land model by computing them
+for each subcomponent, then combining them.
+It takes in 4 arguments related to the land model simulation: `model`, `Y`, `p`, and `t`.
+It also takes in `coupled_atmos`, a ClimaLand.CoupledAtmosphere containing the
+atmospheric properties required for the flux calculation: `u`, `h`,
+`thermal_state`, and `gustiness`.
+
+This function updates each subcomponent in place with its own turbulent fluxes, and also
+returns the total turbulent fluxes for the land model.
+
+Note that this function is currently allocating. We should rewrite it/allocate
+space so that it can update fields in-place instead.
+"""
+function coupler_land_turbulent_fluxes!(p, model::ClimaLand.LandModel, Y, t, coupled_atmos::ClimaLand.CoupledAtmosphere)
```

Comment:
This function is also allocating the combined flux fields. Does it need to return them as a named tuple? Or can it update the named tuple fields in place? 

It also feels like this should be a new method of `ClimaLand.coupler_compute_turbulent_fluxes!` (esp if it is updating something in place)

Reply:
We can update the named tuple fields in place. I wasn't sure of the best way to structure this since we need to get the land fluxes and multiply them by the `area_fraction`. I wanted this function to be independent of ClimaCoupler ideas (e.g. the coupler fields and `area_fraction`, but you're right that it's allocating. Let me try to rewrite this in a clean way without allocating